api/src/main/java/org/apache/iceberg/actions/RewriteDeleteFiles.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import org.apache.iceberg.expressions.Expression;
+
+/**
+ * An action for rewriting delete files according to a rewrite strategy.
+ * Generally used for optimizing the sizing and layout of delete files within a table.
+ */
+public interface RewriteDeleteFiles extends SnapshotUpdate<RewriteDeleteFiles, RewriteDeleteFiles.Result> {
+
+  /**
+   * Convert the equality deletes to the position deletes.
+   *
+   * @return this for method chaining
+   */
+  RewriteDeleteFiles convertEqualityDeletes();
```

Comment:
According to the proposal, this would be a different action because it requires different analysis. Can you remove this? I think that the only strategy should be to rewrite position deletes. I'm not sure if "combine" is the right term there, but it's okay for now.

Reply:
I was waiting for the proposal to be finalized. Now it looks like almost there.  I just updated them.